Parser/parser.c

```diff
@@ -260,15 +261,20 @@ PyParser_AddToken(parser_state *ps, int type, char *str,
                     /* Push non-terminal */
                     int nt = (x >> 8) + NT_OFFSET;
                     int arrow = x & ((1<<7)-1);
-                    dfa *d1 = PyGrammar_FindDFA(
+                    dfa *d1;
+                    if (nt == func_body_suite && !(ps->p_flags & PyCF_TYPE_COMMENTS)) {
+                        D(printf(" [switch func_body_suite to suite]"));
```

Comment:
I would leave a short comment here explaining the motivation.

Reply:
Done.